crates/polars-expr/src/expressions/apply.rs

```diff
@@ -170,7 +170,7 @@ impl ApplyExpr {
                 // })?
                 let out: ListChunked = POOL.install(|| iter.collect::<PolarsResult<_>>())?;
 
-                debug_assert_eq!(out.dtype(), &DataType::List(Box::new(dtype)));
+                debug_assert_eq!(&dtype, out.dtype());
```

Comment:
I think the output field of the expression should created in an aggregation-context-aware manner, so I've updated this assertion.